core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/PhysicalLink.scala

```diff
@@ -41,65 +41,65 @@ object PhysicalLink {
       inputPort: Int,
       partitionInfo: PartitionInfo
   ): PhysicalLink = {
-    val partitionings: Array[(Partitioning, Array[ActorVirtualIdentity])] = partitionInfo match {
+    val partitionings: List[(Partitioning, List[ActorVirtualIdentity])] = partitionInfo match {
       case HashPartition(hashColumnIndices) =>
-        fromPhysicalOp.identifiers
+        fromPhysicalOp.getWorkerIds
           .map(_ =>
             (
               HashBasedShufflePartitioning(
                 defaultBatchSize,
-                toPhysicalOp.identifiers,
+                toPhysicalOp.getWorkerIds,
                 hashColumnIndices
               ),
-              toPhysicalOp.identifiers
+              toPhysicalOp.getWorkerIds
             )
           )
 
       case RangePartition(rangeColumnIndices, rangeMin, rangeMax) =>
-        fromPhysicalOp.identifiers
+        fromPhysicalOp.getWorkerIds
           .map(_ =>
             (
               RangeBasedShufflePartitioning(
                 defaultBatchSize,
-                toPhysicalOp.identifiers,
+                toPhysicalOp.getWorkerIds,
                 rangeColumnIndices,
                 rangeMin,
                 rangeMax
               ),
-              toPhysicalOp.identifiers
+              toPhysicalOp.getWorkerIds
             )
           )
 
       case SinglePartition() =>
-        assert(toPhysicalOp.numWorkers == 1)
-        fromPhysicalOp.identifiers
+        assert(toPhysicalOp.parallelizable == false)
```

Comment:
Previously when I used "single partition" on a UDF input port, the UDF could work. Now it does not work because it failed this assertion. Is this intended?

Reply:
it is not intended. the paralleliable setting for such a case is incorrect. I've fixed it.